### DISCUSSION TITLE
Upsert an array of Entities?

### DISCUSSION THREAD

**Discussion by chanon** [mikro-orm/mikro-orm#2089] [Q&A]
**Is your feature request related to a problem? Please describe.**
Suppose I have an array of plain javascript objects from outside input.
I want to insert any that are missing in the table and update any that already exists (upsert).
I have tried converting the javascript objects to Entities using assign, but when flushing they aren't saved to db.
I have also tried using persist() but then for the items that already exist results in duplicate pk error.
I also tried using merge() .. not sure if it is relevant but then for new items they don't get inserted.

Currently it seems I have to know whether each one already exists or not and call persist or assign accordingly.

TypeORM allows me to just call save on the whole array.

Maybe this is already possible but I just don't know how?

**Describe the solution you'd like**
An easy way to upsert an array of javascript objects or Entities.

For postgresql could use on conflict to handle the upsert.

**Comment by B4nan** [comment]
You need to use QB for upserts, there you have the on conflict support.

**Comment by chanon** [comment]
Is it possible to batch QB requests? otherwise for an array of 500 items, I would be awaiting each qb.execute

**Comment by chanon** [comment]
Or, how can I solve the original problem using existing EntityManager methods?

**Comment by B4nan** [comment]
You can use batch insert with QB. There is no way via UoW other than trying to load the entities first, using explicit transaction so the select is done in the same transaction. 

I have no plans to support native upserts via UoW, if you want those, QB is the only way.

**Comment by B4nan** [comment]
Check the QB tests for examples, for multiinsert just pass array of objects into qb.insert()

https://github.com/mikro-orm/mikro-orm/blob/master/tests/QueryBuilder.test.ts
